components/motor/gpio/motor_encoder.go

```diff
@@ -286,12 +286,12 @@ func (m *EncodedMotor) directionMovingInLock() float64 {
 // SetPower sets the percentage of power the motor should employ between -1 and 1.
 // Negative power implies a backward directional rotational.
 func (m *EncodedMotor) SetPower(ctx context.Context, powerPct float64, extra map[string]interface{}) error {
+	m.opMgr.CancelRunning(ctx)
 	if m.rpmMonitorDone != nil {
 		m.rpmMonitorDone()
 	}
 	if m.loop != nil {
-		m.loop.Stop()
-		m.loop = nil
+		m.loop.Pause()
```

Comment:
o I like this change. Alot cleaner than having to recreate the loop after each run